gtk4/src/expression.rs

```diff
@@ -182,6 +182,56 @@ impl glib::value::ToValueOptional for Expression {
     }
 }
 
+/// Trait containing convenience methods in creating [`gtk::PropertyExpression`] that
+/// looks up a property of a [`glib::Object`].
+pub trait PropExpr {
+    /// Create an expression looking up an object's property
+    fn property_expression(&self, property_name: &str) -> Expression;
+
+    /// Create an expression looking up an object's property with a weak reference.
+    fn weak_property_expression(&self, property_name: &str) -> Expression;
+}
+
+impl<T: IsA<glib::Object>> PropExpr for T {
+    fn property_expression(&self, property_name: &str) -> Expression {
+        let obj_expr = crate::ConstantExpression::new(self).upcast();
+        crate::PropertyExpression::new(T::static_type(), Some(&obj_expr), property_name).upcast()
+    }
+
+    fn weak_property_expression(&self, property_name: &str) -> Expression {
+        let obj_expr = crate::ObjectExpression::new(self).upcast();
+        crate::PropertyExpression::new(T::static_type(), Some(&obj_expr), property_name).upcast()
+    }
+}
+
+/// Trait containing convenience methods to create chains of [`gtk::PropertyExpression`]
+/// or [`gtk::ClosureExpression`].
+pub trait ChainExpr {
```

Comment:
`ExpressionChainExt` maybe? But this could both also be directly added into an `impl Expression` block with exactly the same behaviour without having to define a new trait, or not?

Reply:
I think that could work too, but `closure_expression` and `property_expression` as method name is not pretty clear without docs. At least with the trait, it is signaled by importing the trait

Reply:
The trait would be part of the prelude though.

So maybe we need different method names here?

Reply:
what about `Expression::chain_property` and `Expression::chain_closure`?

Reply:
I like that. @bilelmoussaoui ?

Reply:
Looks good to me as well

Reply:
Renamed it to chain_property and chain_closure